kubespawner/spawner.py

```diff
@@ -403,18 +403,14 @@ def _namespace_default(self):
         minlen=0,
         config=True,
         help="""
-        The command used for starting the single-user server.
+        The command used to start the single-user server.
 
-        Provide either a string or a list containing the path to the startup script command. Extra arguments,
-        other than this path, should be provided via `args`.
+        Either
+          - a string containing a single command or path to a startup script
+          - a list of the command and arguments
+          - `None` (default) to use the Docker image's `CMD`
 
-        This is usually set if you want to start the single-user server in a different python
-        environment (with virtualenv/conda) than JupyterHub itself.
-
-        Some spawners allow shell-style expansion here, allowing you to use environment variables.
-        Most, including the default, do not. Consult the documentation for your spawner to verify!
-
-        If set to `None`, Kubernetes will start the `CMD` that is specified in the Docker image being started.
+        `cmd` will be passed to the image's `ENTRYPOINT`.
```

Comment:
Is this what you mean?
```suggestion
        If `cmd` is set it will be passed to the image's `ENTRYPOINT` along with any additional arguments.
```

Reply:
```suggestion
        If `cmd` is set, it will be augmented with `spawner.get_args()` and passed to the spawned k8s Pod's container's `args` field that overrides the Dockerfile's `CMD` section. This will leave the Dockerfile's `ENTRYPOINT` section intact, and there is currently no way to specify the k8s Pod's Container's `command` which would override the Dockerfile's `ENTRYPOINT`.
```

I think something like this. This is a messy story but it is an accurate representation of the current situation.  `get_args()` is [defined in the JupyterHub Spawner base class](https://github.com/jupyterhub/jupyterhub/blob/3fec19d191119e4203c243f44c43e74c8fe62f04/jupyterhub/spawner.py#L995).

Reply:
Instead of bringing in the internal K8s details how about keep it simple? 
```suggestion
        If `cmd` is set, it will be augmented with `spawner.get_args(). This will override the `CMD` specified in the Docker image.
```
As you point out it's already complicated, and now I think it'd be better not to mention entrypoint, args or command. If we add entrypoint I think we could also keep it's help doc simple, and instead add the complicated explanation to the general docs instead, e.g. on the [overview page](https://jupyterhub-kubespawner.readthedocs.io/en/latest/overview.html) or a new customisation guide page? That way you can include examples without bulking out the API docs too much.